game-core/src/system/ally.rs

```diff
@@ -22,60 +83,34 @@ pub struct Grouper;
 impl<'s> System<'s> for Grouper {
     type SystemData = (
         ReadStorage<'s, Ally>,
-        WriteStorage<'s, Transform>,
-        WriteStorage<'s, Parent>,
-        WriteStorage<'s, Player>,
+        ReadStorage<'s, Transform>,
+        WriteStorage<'s, Motion>,
+        ReadStorage<'s, Player>,
         Entities<'s>,
     );
 
-    fn run(
-        &mut self,
-        (allies, mut transforms, mut parents, mut players, entities): Self::SystemData,
-    ) {
-        let get_relative_position = |mut num_allies: u32| {
-            num_allies %= 9;
-
-            let ind = if num_allies == 4 {
-                num_allies + 1 // skip player position
-            } else {
-                num_allies
-            } as i32;
-
-            let row = ind / 3 - 1;
-            let col = ind % 3 - 1;
-
-            (row as f32 * 32.0, col as f32 * 32.0)
-        };
-
-        let (p_transform, allies_count, p_entity) = {
-            let (t, p, e) = (&transforms, &players, &entities)
+    fn run(&mut self, (allies, transforms, mut motions, players, entities): Self::SystemData) {
+        let p_transform = {
+            let (t, _) = (&transforms, &players)
                 .join()
                 .next()
                 .expect("no player found");
-            (t.clone(), p.num_allies, e)
+            (t.clone())
         };
 
-        let mut orphans = vec![];
-
-        for (_, _, transform, entity) in (&allies, !&parents, &mut transforms, &entities).join() {
+        let mut merged = vec![];
+        for (_ally, transform, e, _) in (&allies, &transforms, &*entities, !&motions).join() {
```

Comment:
I'm assuming that this `!&motion` will eventually be replaced by `!&parent` and allies will be constructed with a motion component by default once we add the position editor.

Reply:
possibly. it depends on what we want the behavior to be after the position editor. In some ways, dealing with allies outside of the parent as the translations may not allow this to be that clear.